ori/volume/server/volume_create.go

```diff
@@ -17,260 +17,132 @@ package server
 	}
 
-	return size, nil
+	return nil
 }
 
-func (s *Server) getVolumeConfig(ctx context.Context, volume *ori.Volume) (*AggregateCephVolume, error) {
-	volumeClass := &storagev1alpha1.VolumeClass{}
-	if err := s.client.Get(ctx, types.NamespacedName{Name: volume.Spec.Class}, volumeClass); err != nil {
-		return nil, fmt.Errorf("unable to get VolumeClass: %w", err)
+func (s *Server) validateImage(ctx context.Context, log logr.Logger, volume *ori.Volume) (*PopulationImage, error) {
+	imageName := volume.GetSpec().GetImage()
+	if imageName == "" {
+		return nil, nil
 	}
 
-	pvc := &corev1.PersistentVolumeClaim{
-		TypeMeta: metav1.TypeMeta{
-			Kind:       "PersistentVolumeClaim",
-			APIVersion: "v1",
-		},
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      s.idGen.Generate(),
-			Namespace: s.namespace,
-		},
-		Spec: corev1.PersistentVolumeClaimSpec{
-			AccessModes: []corev1.PersistentVolumeAccessMode{corev1.ReadWriteOnce},
-			Resources: corev1.ResourceRequirements{
-				Requests: corev1.ResourceList{corev1.ResourceStorage: *resource.NewQuantity(int64(volume.Spec.Resources.StorageBytes), resource.DecimalSI)},
-			},
-			VolumeMode:       func(m corev1.PersistentVolumeMode) *corev1.PersistentVolumeMode { return &m }(corev1.PersistentVolumeBlock),
-			StorageClassName: &s.rookPoolStorageClass,
-		},
-	}
-
-	var imagePvc *corev1.PersistentVolumeClaim
-	if volume.Spec.Image != "" {
-		imageName := GetSanitizedImageName(volume.Spec.Image)
-		apiutils.SetImageAnnotation(pvc, volume.Spec.Image)
-
-		pvc.Spec.DataSourceRef = &corev1.TypedObjectReference{
-			APIGroup: pointer.String("snapshot.storage.k8s.io"),
-			Kind:     "VolumeSnapshot",
-			Name:     imageName,
-		}
-
-		size, err := getImageSize(ctx, volume.Spec.Image)
-		if err != nil {
-			return nil, fmt.Errorf("unable to get image size: %w", err)
-		}
-
-		imagePvc = &corev1.PersistentVolumeClaim{
-			TypeMeta: metav1.TypeMeta{
-				Kind:       "PersistentVolumeClaim",
-				APIVersion: "v1",
-			},
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      imageName,
-				Namespace: s.namespace,
-			},
-			Spec: corev1.PersistentVolumeClaimSpec{
-				AccessModes: []corev1.PersistentVolumeAccessMode{corev1.ReadWriteOnce},
-				Resources: corev1.ResourceRequirements{
-					Requests: corev1.ResourceList{corev1.ResourceStorage: *size},
-				},
-				VolumeMode:       func(m corev1.PersistentVolumeMode) *corev1.PersistentVolumeMode { return &m }(corev1.PersistentVolumeBlock),
-				StorageClassName: pointer.String(s.rookPoolStorageClass),
-				//set DataSourceRef that populator picks up the pvc
-				DataSourceRef: &corev1.TypedObjectReference{
-					APIGroup: pointer.String(storagev1alpha1.SchemeGroupVersion.String()),
-					Kind:     "Volume",
-					Name:     pvc.Name,
-				},
-			},
-		}
-		metautils.SetLabel(imagePvc, volumev1alpha1.ManagerLabel, volumev1alpha1.VolumeCephletManager)
-
+	log.V(2).Info("Resolving os image")
+	img, err := s.populator.ResolveImage(ctx, imageName)
```

Comment:
Validating should not include resolving the image. A create with an image that is not yet available should succeed since e.g. the push to the registry might happen at a later point in time. For comparison, see e.g. image pulling in pods in k8s. A pod's image might not be there upon creation of the `corev1.Pod`, but the container runtime retries indefinitely (unless the pod is deleted in the meantime).